tls/s2n_client_hello.c

```diff
@@ -111,6 +111,23 @@ ssize_t s2n_client_hello_get_cipher_suites_length(struct s2n_client_hello *ch) {
     return ch->cipher_suites.size;
 }
 
+int s2n_client_hello_cb_done(struct s2n_connection *conn)
+{
+    struct s2n_client_hello *client_hello;
+
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(conn->config);
+    POSIX_ENSURE(conn->config->client_hello_cb_mode ==
+        S2N_CLIENT_HELLO_CB_NONBLOCKING, S2N_ERR_INVALID_STATE);
+    POSIX_GUARD_PTR(client_hello = s2n_connection_get_client_hello(conn));
```

Comment:
What's the difference between `POSIX_ENSURE_REF` and `POSIX_GUARD_PTR`?

Reply:
Also maybe a `S2N_ERR_INVALID_STATE` is a better error here, as it likely indicates that we haven't parsed ClientHello yet.

Reply:
ENSURE_REF will override the errno with  `S2N_ERR_NULL` so GUARDing is usually better if the null reference came from a fallible function call.

Reply:
Also maybe a S2N_ERR_INVALID_STATE is a better error here, as it likely indicates that we haven't parsed ClientHello yet.
> makes sense. I will replace this with a check to ensure we have parsed the client hello